src/Components/AppDownloadFooter.tsx

```diff
@@ -37,20 +39,36 @@ export const AppDownloadFooter: FC<AppDownloadFooterProps> = () => {
   const ref = useRef<HTMLDivElement | null>(null)
 
   const { downloadAppUrl } = useDeviceDetection()
+  const { match } = useRouter()
 
-  useRouteComplete({
-    // Used to ensure we just show the banner after an initial internal navigation
-    onComplete: async () => {
-      if (!ref.current || !("animate" in ref.current)) return
+  const isVisible = useRef(false)
+  const routeChanges = useRef(-1)
+
+  useEffect(() => {
+    const showBanner = async () => {
+      if (!ref.current || !("animate" in ref.current) || isVisible.current) {
+        return
+      }
 
       const animation = ref.current.animate(
         [{ transform: "translateY(100%)" }, { transform: "translateY(0)" }],
         { duration: 250, easing: "ease-out", fill: "forwards" }
       )
 
       await animation.finished
-    },
-  })
+      isVisible.current = true
+    }
+
+    // Increment route changes counter when pathname changes
+    if (match.location.pathname) {
+      routeChanges.current++
+    }
+
+    // Show banner after second route change
+    if (routeChanges.current === 2) {
```

Comment:
if the `match.elements` approach doesnt work, could we roll this up into a reusable hook? Or is the `=== 2` bit specific to this page?

Reply:
It indeed no longer works and this is specific to this component